packages/modules/viewer-react/src/services/telemetry/TelemetryService.ts

```diff
@@ -21,20 +37,30 @@ class TelemetryService implements TelemetryClient {
     }
 
     try {
-      // TODO 3.0 how to get user info?
-      // const token = await ViewerAuthorization.client.
-      // const accessToken =
-      //   (await ViewerAuthorization.client.getAccessToken()) as AccessToken;
-      // const user = accessToken?.getAccessToken().getUserInfo();
-      // if (user && accessToken) {
-      //   this._appInsights.setAuthenticatedUserContext(
-      //     user.id,
-      //     user.organization?.id,
-      //     true
-      //   );
-      // } else {
-      //   this._appInsights.clearAuthenticatedUserContext();
-      // }
+      const accessToken = await ViewerAuthorization.client?.getAccessToken();
+      if (accessToken) {
+        // https://developer.bentley.com/apis/users/operations/me/
+        const url = `https://${
+          process.env.IMJS_URL_PREFIX ?? ""
+        }api.bentley.com/users/me`;
+        const resp = await fetch(url, {
+          headers: {
+            Authorization: accessToken,
+          },
+        });
+        if (resp.ok) {
+          const { user }: UserInfo = await resp.json();
+          this._appInsights.setAuthenticatedUserContext(
```

Comment:
the old api using introspection on the actual jwt token would return ids we could use, here these arent obfuscated. i think itsok to collect these, especially since the user has to grant permission to use this api, but wanted to check with you guys

Reply:
I'm not sure if we should be tracking it in this way. Plus it won't have the desired effect if someone is using a access token that's client creds. Not sure how the API would handle that case.

Reply:
Yea that why i said in the nextversion.md its optional because of the client-cred workflow. id imagine it throws an error in the client-cred workflow, so itd be tracked as un authenticated user

Reply:
updated to introspect the token client side, and now the ids are obfuscated a bit more